File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution(object):
    def reversePrefix(self, word, ch):
        if ch not in word:
            return word
        
        x = word.index(ch)
        rev = word[:x + 1][:: -1]
        word = word.replace(word[:x + 1], rev, 1)

        return word

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        if (!word.contains(String.valueOf(ch))) {
            return word;
        }
        
        int x = word.indexOf(ch);
        String rev = new StringBuilder(word.substring(0, x + 1)).reverse().toString();
        return word.replaceFirst(word.substring(0, x + 1), rev);
    }
}

